### PR TITLE
! 9.0 BUGFIX: fix content stream cleanup

### PR DESCRIPTION
**Pull request by skurfuerst**
This is needed for Asset Usage to work properly. - https://github.com/neos/neos-development-collection/pull/4088 

## How to migrate

```bash
# to migrate the renaming in the projection
./flow cr:setup

# to ensure we do not have any leftover dangling content streams
./flow contentStream:prune --removeTemporary
```

### CODE REVIEW

File: Neos.ContentRepository.Core/Classes/Feature/WorkspaceCommandHandler.php [hunk 1]
```diff
@@ -545,15 +546,19 @@ private function handlePublishIndividualNodesFromWorkspace(
         // 6) switch content stream to forked WS.
         // if we got so far without an Exception, we can switch the Workspace's active Content stream.
         $streamName = StreamName::fromString('Workspace:' . $command->workspaceName);
-        $events = Events::with(
+        $events = Events::fromArray([
             new WorkspaceWasPartiallyPublished(
                 $command->workspaceName,
                 $baseWorkspace->workspaceName,
                 $remainingContentStream,
-                $workspace->currentContentStreamId,
```
Comment (skurfuerst) [inline Neos.ContentRepository.Core/Classes/Feature/WorkspaceCommandHandler.php]: how did this EVER work without a type error???

File: Neos.ContentRepository.Core/Classes/Feature/WorkspacePublication/Event/WorkspaceWasPartiallyPublished.php [hunk 2]
```diff
@@ -38,9 +38,9 @@ public function __construct(
          */
         public readonly ContentStreamId $newSourceContentStreamId,
         /**
-         * The old content stream, which contains ALL the data (discarded and non-discarded)
+         * The old content stream, which contains ALL the data (ediscarded and non-discarded)
          */
-        public readonly ContentStreamId $previousSourceContentStreamId,
+        // => "ContentStreamWasReleased" event ==> public readonly ContentStreamId $previousSourceContentStreamId,
```
Comment (skurfuerst) [inline Neos.ContentRepository.Core/Classes/Feature/WorkspacePublication/Event/WorkspaceWasPartiallyPublished.php]: uh, why did I remove this?

File: Neos.Neos/Classes/PendingChangesProjection/ChangeProjection.php [hunk 3]
```diff
@@ -88,6 +88,18 @@ private function setupTables(): void
             throw new \RuntimeException('Failed to retrieve Schema Manager', 1625653914);
         }
 
+        $currentSchema = $schemaManager->createSchema();
```
Comment (skurfuerst) [inline Neos.Neos/Classes/PendingChangesProjection/ChangeProjection.php:92]: TODO: separate into extra PR

File: Neos.ContentRepository.Core/Classes/Feature/WorkspaceCommandHandler.php (hunk 1 above)
Comment (skurfuerst) [reply Neos.ContentRepository.Core/Classes/Feature/WorkspaceCommandHandler.php]: found it.

File: Neos.ContentRepository.Core/Classes/Projection/ContentStream/ContentStreamFinder.php [hunk 4]
```diff
@@ -25,14 +25,84 @@
...
+     */
     public const STATE_CREATED = 'CREATED';
+
+    /**
+     * STATE_FORKED means the content stream was forked from an existing content stream, but not yet assigned
+     * to a workspace.
+     *
+     * **temporary state** which should not appear if the system is idle (for content streams which are used with workspaces).
+     */
+    public const STATE_FORKED = 'FORKED';
```
Comment (skurfuerst) [inline Neos.ContentRepository.Core/Classes/Projection/ContentStream/ContentStreamFinder.php:90]: TODO: migration => REBASING => FORKED?

**Review by bwaidelich — COMMENTED** [review]
Nice one, makes sense so far!

As discussed, we should probably also remove `state` from the Content Stream Read Model and move it to (a separate table?) of the Workspace projection (since the notion of a CS state only makes sense in conjunction with Workspaces and the CR works without them)

**Comment by skurfuerst** [comment]
@bwaidelich I would rather do this change somewhen later instead of now - as this will greatly increase what needs to be done here...

File: Neos.Neos/Classes/PendingChangesProjection/ChangeProjection.php (hunk 3 above)
Comment (skurfuerst) [reply Neos.Neos/Classes/PendingChangesProjection/ChangeProjection.php:92]: (done into extra commit)

**Pull request merged**